### DISCUSSION TITLE
服务创建失败了为啥还把queue添加到全局消息队列?

### DISCUSSION THREAD

**Discussion by xiaoqinguo** [cloudwu/skynet#1692] [General]
skynet_context_new函数：
![}5D T9J(DMAS0){9OMXODSY](https://user-images.githubusercontent.com/9319187/211517902-021764da-667d-4bae-9fa2-0d1862b1a6ec.png)
![NPR}(_OBNHC`TKI)4E3HIKQ](https://user-images.githubusercontent.com/9319187/211517979-e2c59b3b-92c9-4fe4-bd14-bdf3e2c96a8a.png)
q->release初始化为0，当服务创建失败时，context都会delete掉，这里为啥还把queue添加到全局消息队列？

**Comment by cloudwu** [comment]
skynet_mq_release 并非设计成只在服务创建失败这一种情况下被调用。

**Reply by xiaoqinguo** [reply]
skynet_mq_release的设计我没有疑问，但是创建失败还把queue添加到全局消息队列不会有问题吗？我今天初看代码，可能还有理解不到位的地方，如果确定没有问题那我就再看看吧
